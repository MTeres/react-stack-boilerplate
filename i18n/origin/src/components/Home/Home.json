[
  {
    "id": "Home.tittle",
    "description": "Tittle",
    "defaultMessage": "Hey you are at home, locale: {locale}"
  }
]