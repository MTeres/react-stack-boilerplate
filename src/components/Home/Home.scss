.Home{
  background: linear-gradient(20deg,blue,rgb(219,112,147));
}
