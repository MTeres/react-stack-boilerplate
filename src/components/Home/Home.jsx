/* ------------------------------------------
   Home - Component
--------------------------------------------- */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Switch,
  Route,
  Redirect,
} from 'react-router-dom';
import { injectIntl, defineMessages } from "react-intl";
import styled, { css } from 'styled-components';
import LeftComponent from '../LeftComponent/LeftComponent';
import RightComponent from '../RightComponent/RightComponent';
import './Home.scss';

const messages = defineMessages({
  tittle: {
    id: 'Home.tittle',
    defaultMessage: 'Hey you are at home, locale: {locale}',
    description: 'Tittle',
  },
});

class Home extends Component {
  static displayName = 'Home';
  static propTypes = {
    intl: PropTypes.object.isRequired,
    updateCurrentLang: PropTypes.func.isRequired,
  };
  render() {
    const { intl: { locale, formatMessage }, updateCurrentLang } = this.props;
    return (
      <Main className="Home">
        <p>{formatMessage(messages.tittle, { locale: locale })}</p>
        <Switch>
          <Route component={LeftComponent} exact path="/home/left" />
          <Route component={RightComponent} exact path="/home/right" />
          <Redirect from="/home" to="/home/left" />
        </Switch>
        <SectionButtons>
          <LocaleButton
            active={locale === 'pt'}
            onClick={() => updateCurrentLang('pt')}
          >
            pt
          </LocaleButton>
          <LocaleButton
            active={locale === 'en'}
            onClick={() => updateCurrentLang('en')}
          >
            en
          </LocaleButton>
          <LocaleButton
            active={locale === 'fr'}
            onClick={() => updateCurrentLang('fr')}
          >
            fr
          </LocaleButton>
        </SectionButtons>
      </Main>
    );
  }
}

// styled components
const Main = styled.div`
  padding: 20px;
  display: block;
  min-height: 100vh;
  > p {
    font-weight: 300;
    color: #fff;
    text-transform: uppercase;
    text-align: center;
    font-size: 20px;
  }
`;

const SectionButtons = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 25%;
  margin: 20px auto;
`;

const LocaleButton = styled.div`
    color: black;
    text-transform: uppercase;
    text-align: center;
    font-size: 20px;
    padding: 15px;
    border-radius: 100%;
    background-color: papayawhip;
    border: 1px solid papayawhip;
    cursor: pointer;
  ${p => p.active && css`
    display: none;
  `};
`;

export default injectIntl(Home);
