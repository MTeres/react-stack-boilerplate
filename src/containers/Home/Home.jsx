/* ------------------------------------------
   Home - Component
--------------------------------------------- */
import React, { Component } from 'react';
import { connect } from "react-redux";
import PropTypes from 'prop-types';
import { updateCurrentLang } from '../../state/modules/i18n';
import Home from '../../components/Home/Home';

const mapStateToProps = () => ({});

const bindActions = { updateCurrentLang };

const decorator = connect(mapStateToProps, bindActions);


class HomeContainer extends Component {
  static displayName = 'Home';
  static propTypes = {
    updateCurrentLang: PropTypes.func.isRequired,
  };
  render() {
    const { updateCurrentLang } = this.props;
    return (<Home updateCurrentLang={updateCurrentLang} />);
  }
}

export default decorator(HomeContainer);
