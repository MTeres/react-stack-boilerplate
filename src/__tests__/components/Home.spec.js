import React from 'react';
import path from 'path';
import renderer from 'react-test-renderer';
import Home from '../../components/Home/Home';
import { StaticRouter } from 'react-router-dom';

const context = {};

describe('<Home />', () => {
  const snapshotFileName = path.join(__dirname, 'Home.spec.js.snap');
  test('Link matches snapshot', () => {
    const component = renderer.create(
      <StaticRouter context={context} location="/home/left">
        <Home />
      </StaticRouter>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot(snapshotFileName, 'home-with-route');
  });
});
