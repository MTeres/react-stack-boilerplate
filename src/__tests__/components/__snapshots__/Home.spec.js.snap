// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Link matches snapshot: /home/chance-01/dev/react-starter/src/__tests__/components/Home.spec.js.snap 1`] = `
<div
  className="Home sc-ifAKCX bWQfHe"
>
  <p>
    We are at home :)
  </p>
  <div
    className="sc-bdVaJa hXNufC"
  >
    <p
      className="sc-bwzfXH cXGpPr"
    >
      LeftSide :)
    </p>
    <p
      className="sc-bwzfXH edTkHj"
    >
      <a
        href="/home/right"
        onClick={[Function]}
      >
        click
      </a>
    </p>
  </div>
</div>
`;
